### PR TITLE
Use WithObjectMetadata in ComposeObject().

### PR DESCRIPTION
**Pull request by coryan**
The ObjectMetadata parameter should be optional for this function.

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1519)

### CODE REVIEW

File: google/cloud/storage/internal/object_requests.h [hunk 1]
```diff
@@ -263,28 +263,21 @@ std::ostream& operator<<(std::ostream& os, UpdateObjectRequest const& r);
...
+                                std::vector<ComposeSourceObject> source_objects,
+                                std::string destination_object_name);
 
   /// Returns the request as the JSON API payload.
-  std::string json_payload() const { return json_payload_; }
-
-  ObjectMetadata const& destination_metadata() const {
-    return destination_metadata_;
-  }
+  std::string JsonPayload() const;
```
Comment (frankyn) [inline google/cloud/storage/internal/object_requests.h]: Ah, is this a style guide inconsistency?

Comment (frankyn) [reply google/cloud/storage/internal/object_requests.h]: Thanks! I need to be more careful next time 😄

Comment (coryan) [reply google/cloud/storage/internal/object_requests.h]: Oh, you were doing the right thing.  See the C++ style guide recommends (and some say require) `snake_case` for simple accessors, but it requires `PascalCase` for more complex functions.  So when the implementation changes the naming convention changes.  That is a terrible terrible idea in the style guide.  It should be burnt with fire.

Comment (frankyn) [reply google/cloud/storage/internal/object_requests.h]: ahhhh. oh wow. that's a loaded style guide requirement. I guess that makes sense, but at the same time confusing. Thanks for clarifying.

**Pull request merged**